include/crocoddyl_msgs/whole_body_state_publisher.h

```diff
@@ -137,6 +137,67 @@ class WholeBodyStateRosPublisher {
     }
   }
 
+  /**
+   * @brief Publish a whole-body state ROS message
+   *
+   * @param t[in]    Time in secs
+   * @param q[in]    Configuration vector (dimension: model.nq)
+   * @param v[in]    Generalized velocity (dimension: model.nv)
+   * @param a[in]    Generalized acceleration (dimension: model.nv)
+   * @param tau[in]  Joint effort (dimension: model.nv)
+   * @param p[in]    Contact position
+   * @param pd[in]   Contact velocity
+   * @param f[in]    Contact force, type and status
+   * @param s[in]    Contact surface and friction coefficient
+   */
+  void
+  publish(const double t, const Eigen::Ref<const Eigen::VectorXd> &q,
+          const Eigen::Ref<const Eigen::VectorXd> &v,
+          const Eigen::Ref<const Eigen::VectorXd> &a,
+          const Eigen::Ref<const Eigen::VectorXd> &tau,
+          const std::map<std::string, pinocchio::SE3> &p = DEFAULT_SE3,
+          const std::map<std::string, pinocchio::Motion> &pd  = DEFAULT_MOTION,
+          const std::map<std::string, std::tuple<pinocchio::Force, ContactType,
+                                                 ContactStatus>> &f = DEFAULT_FORCE,
+          const std::map<std::string, std::pair<Eigen::Vector3d, double>> &s = DEFAULT_FRICTION) {
+    if (pub_.trylock()) {
+      pub_.msg_.header.frame_id = odom_frame_;
+      if (is_reduced_model_) {
+        fromReduced(model_, reduced_model_, qfull_, vfull_, afull_, ufull_, q, v, a, tau,
+                    qref_, joint_ids_);
+        crocoddyl_msgs::toMsg(model_, data_, pub_.msg_, t, qfull_,
+                              vfull_, afull_, ufull_, p, pd, f, s);
+      } else {
+        crocoddyl_msgs::toMsg(model_, data_, pub_.msg_, t, q, v, a, tau, p, pd,
+                              f, s);
+      }
+      pub_.unlockAndPublish();
+    }
+  }
+  
+  /**
+   * @brief updates the inertial parameters of the pinocchio model
+   *
+   * @param body_name[in] name of the desired body to update the inertial parameters
+   * @param psi[in]       Vector containing the inertial parameters
+   */
+  void update_model_inertial_parameters(const std::string &body_name, const Eigen::Ref<const Eigen::VectorXd> &psi){
```

Comment:
The function name is misleading, rename it to
```suggestion
  void update_body_inertial_parameters(const std::string &body_name, const Eigen::Ref<const Eigen::VectorXd> &psi){
```

Please take care of the other cases as well.